TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js

```diff
@@ -53,22 +50,33 @@
         }
 
         function runRun() {
-            $http.put('/api/v1/Runs/' + $stateParams.id, {
-                envFull: vm.run.envFull,
-                build: vm.run.build
-            }).success(function (run) {
-                RunsApiService.getTestsOfRun(run._id).query().$promise
-                    .then(function (tests) {
-                        vm.run.tests = tests;
-                        console.log(vm.run);
+            RunsApiService.getTestsOfRun(vm.run._id).query().$promise
+                .then(function (tests) {
+                    vm.run.tests = JSON.parse(JSON.stringify(tests));
```

Comment:
the main advantage of the resource is the possibility to do as follows:

```js
// Define CreditCard class
var CreditCard = $resource('/user/:userId/card/:cardId',
 {userId:123, cardId:'@id'}, {
  charge: {method:'POST', params:{charge:true}}
 });

// We can retrieve a collection from the server
var cards = CreditCard.query(function() {
  // GET: /user/123/card
  // server returns: [ {id:456, number:'1234', name:'Smith'} ];

  var card = cards[0];
  card.name = "J. Smith";
  // non GET methods are mapped onto the instances
  card.$save();
  // POST: /user/123/card/456 {id:456, number:'1234', name:'J. Smith'}
  // server returns: {id:456, number:'1234', name: 'J. Smith'};
});
```

note the line `card.name = "J. Smith";` and after it we can do `$save` and it might work. without using this feature there is no sense in $resource. basic $http is easier.